optout/src/main/java/gov/cms/ab2d/optout/ConsentConverterServiceImpl.java

```diff
@@ -0,0 +1,104 @@
+package gov.cms.ab2d.optout;
+
+import gov.cms.ab2d.common.model.Consent;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.stereotype.Component;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.Optional;
+import java.util.regex.Pattern;
+
+@Slf4j
+@Component
+public class ConsentConverterServiceImpl implements ConsentConverterService {
+
+    private static final Pattern HEALTH_INSURANCE_CLAIM_NUMBER_PATTERN = Pattern.compile("\\d{9}[A-Za-z0-9]{0,2}");
+
+    private static final int HEALTH_INSURANCE_CLAIM_NUMBER_START = 0;
+    private static final int HEALTH_INSURANCE_CLAIM_NUMBER_END = 11;
+    private static final int EFFECTIVE_DATE_START = 354;
+    private static final int EFFECTIVE_DATE_END = 362;
+    private static final int SOURCE_CODE_START = 362;
+    private static final int SOURCE_CODE_END = 367;
+    private static final int PREF_INDICATOR_START = 368;
+    private static final int PREF_INDICATOR_END = 369;
+
+    private static final String OPT_OUT_INDICATOR = "N";
+
+    @Override
+    public Optional<Consent> convert(String line, int lineNum) {
+        if (isHeader(line)) {
+            log.warn("Skipping Header row");
+            return Optional.empty();
+        }
+
+        if (isTrailer(line)) {
+            log.warn("Skipping Trailer row");
+            return Optional.empty();
+        }
+
+        var sourceCode = line.substring(SOURCE_CODE_START, SOURCE_CODE_END);
+        if (StringUtils.isBlank(sourceCode)) {
+            log.warn("SourceCode is blank. Skipping row : {}", lineNum);
+            return Optional.empty();
+        }
+        if (!sourceCode.trim().matches("1-?800")) {
+            log.warn("Invalid SourceCode: {}. Skipping row : {}", sourceCode, lineNum);
+            throw new RuntimeException("Unexpected beneficiary data sharing source code in row : "  + lineNum);
+        }
+
+        var prefIndicator = line.substring(PREF_INDICATOR_START, PREF_INDICATOR_END);
+        if (!OPT_OUT_INDICATOR.equalsIgnoreCase(prefIndicator)) {
+            // we only care about opt-out records
+            log.info("Preference Indicator is NOT opt-out. It was : {}, Skipping row : {}", prefIndicator, lineNum);
+            return Optional.empty();
+        }
```

Comment:
In the reference implementation, they were including both OPT_OUT and OPT_IN records.
I only included the opt-out rows as I thought we are interested only in the OPT_OUT records. Hope that is okay.